File: okj/rendszerüzemeltetö/snooker/f#/Snooker.fs
open System.IO

[<EntryPoint>]
let main _ =
    let versenyzotKeszit (k: string[]) = {| Helyezes = int k.[0]; Nev = k.[1]; Orszag = k.[2]; Nyeremeny = int k.[3] |}
    let versenyzok = File.ReadLines("snooker.txt") |> Seq.skip(1)
                                                   |> Seq.map(fun k -> k.Split(";") |> versenyzotKeszit)
                                                   |> Seq.toArray

    printfn "3. Feladat: Versenyzők száma: %d" versenyzok.Length

    versenyzok |> Seq.averageBy(fun k -> float k.Nyeremeny)
               |> printfn "4. Feladat: Átlag kereset: %.2f"

    versenyzok |> Seq.filter(fun k -> k.Orszag = "Kína")
               |> Seq.maxBy(fun k -> k.Nyeremeny)
               |> fun k -> printfn "5. Feladat: Legjobban kereső kínai versenyző:\n\tHelyezés: %d\n\tOrszág: Kína\n\tNyeremény: %d FT" k.Helyezes (k.Nyeremeny * 380)

    versenyzok |> Seq.exists(fun k -> k.Orszag = "Norvégia")
               |> fun k -> printfn "6. Feladat: %s norvég játékos" (if k then "Van" else "Nincs")

    printfn "7. Feladat:"

    versenyzok |> Seq.map(fun k -> k.Orszag)
               |> Seq.countBy(id)
               |> Seq.filter(fun (_, v) -> v > 4)
               |> Seq.iter(fun (k, v) -> printfn "%s: %d fő" k v)
    0

File: okj/rendszeruzemelteto/snooker/python/Snooker.py
from statistics import mean
from collections import Counter

class Versenyzo:

    def __init__(self, line: str):
        split = line.split(';')

        self.helyezes = int(split[0])
        self.nev = split[1]
        self.orszag = split[2]
        self.nyeremeny = int(split[3])

with open('snooker.txt') as file:
    lines = file.readlines()
    versenyzok = [ Versenyzo(lines[i]) for i in range(1, len(lines)) ]

print(f'3. Feladat: Versenyzők száma: {len(versenyzok)}')

atlagKereset = mean(k.nyeremeny for k in versenyzok)
print('4. Feladat: Átlag kereset: %.2f' % atlagKereset)

legjobbKinai = max((k for k in versenyzok if k.orszag == 'Kína'), key = lambda k: k.nyeremeny)
print(f'5. Feladat: Legjobban kereső kínai versenyző:\n  Helyezés: {legjobbKinai.helyezes}\n  Név: {legjobbKinai.nev}\n  Nyeremény: {legjobbKinai.nyeremeny * 380} FT')

voltNorveg = any(True for k in versenyzok if k.orszag == 'Norvégia')
print(f'6. Feladat: {"Van" if voltNorveg else "Nincs"} norvég játékos')
print('7. Feladat:')

stat = Counter(k.orszag for k in versenyzok)
for orszag, db in stat.items():
    if db > 4:
        print(f'{orszag}: {db} fő')